proxy/http/HttpConfig.cc

```diff
@@ -900,6 +900,68 @@ register_stat_callbacks()
                      (int)http_sm_finish_time_stat, RecRawStatSyncSum);
 }
 
+static bool
+set_negative_caching_list(const char *name, RecDataT dtype, RecData data, HttpConfigParams *c, bool update)
+{
+  bool ret = false;
+  // values from proxy.config.http.negative_caching_list
+  if (0 == strcasecmp("proxy.config.http.negative_caching_list", name) && RECD_STRING == dtype) {
+    if (data.rec_string) {
+      // parse the list of status code
+      ts::TextView status_list(std::string(data.rec_string));
```

Comment:
Why is a `std::string` used here? Why not `status_list(data.rec_string, strlen(data.rec_string));`? The use of `std::string` in this way already requires a null terminate string.